### PR TITLE
fix seg fault in find_tagfunc_tags

### PR DESCRIPTION
**Pull request by fkm3**
I expect this to fix the crash from https://github.com/neovim/neovim/issues/15221.

### CODE REVIEW

**Comment by fkm3** [comment]
I tried to reproduce with a test, but failed. The below patch is able to cause the out of bound access (I verified by adding a check to the code), but it doesn't seg fault or trigger asan/valgrind errors.

```
diff --git a/src/nvim/testdir/test_tagfunc.vim b/src/nvim/testdir/test_tagfunc.vim
index ffc1d63b9..22828a39f 100644
--- a/src/nvim/testdir/test_tagfunc.vim
+++ b/src/nvim/testdir/test_tagfunc.vim
@@ -117,4 +117,26 @@ func Test_tagfunc_settagstack()
   delfunc Mytagfunc2
 endfunc

+func Test_tagfunc_settagstack_many()
+
+  func Mytagfunc1(pat, flags, info)
+    return [{'name' : 'mytag', 'filename' : 'Xtest', 'cmd' : '1'}]
+  endfunc
+  set tagfunc=Mytagfunc1
+  call writefile([''], 'Xtest')
+
+  for i in range(0,20)
+    let pos = [bufnr()]  + getcurpos()[1:]
+    let newtag = [{'tagname' : 'mytag' + i, 'from' : pos}]
+    call settagstack(1, {'items' : newtag}, 'a')
+    call settagstack(1, {'curidx' : 21})
+  endfor
+
+  tag
+
+  call delete('Xtest')
+  set tagfunc&
+  delfunc Mytagfunc1
+endfunc
```

File: src/nvim/tag.c [hunk 1]
```diff
@@ -1143,7 +1143,15 @@ static int find_tagfunc_tags(char_u *pat, garray_T *ga, int *match_count, int fl
   typval_T args[4];
   typval_T rettv;
   char_u flagString[4];
-  taggy_T *tag = &curwin->w_tagstack[curwin->w_tagstackidx];
+  taggy_T *tag = NULL;
+
+  if (curwin->w_tagstacklen > 0) {
+    if (curwin->w_tagstackidx == curwin->w_tagstacklen) {
+      tag = &curwin->w_tagstack[curwin->w_tagstackidx - 1];
```
Comment (fkm3) [inline src/nvim/tag.c:1150]: not sure if this case make sense. maybe should leave it null instead

**Review by justinmk — APPROVED** [review]
looks reasonable to me. Did you compare `find_tagfunc_tags` in Vim ? cc @jamessan

**Comment by fkm3** [comment]
At vim HEAD, it seems the relevant part of `find_tagfunc_tags` is identical. I haven't tried to really comprehend the code, so I'm not sure if it is upholding different pre-conditions. I haven't tried to repro in vim either. Maybe I can send the patch upstream and see what they think.

**Pull request merged**